app/services/events/EventsService.scala

```diff
@@ -95,4 +97,15 @@ trait EventsService {
   def getFsbTypes: Future[List[FsbType]] = eventsConfigRepo.fsbTypes
 
   def getTelephoneInterviewTypes: Future[List[TelephoneInterviewType]] = eventsConfigRepo.telephoneInterviewTypes
+
+  def findSchemeByEvent(eventId: String): Future[Option[Scheme]] = {
+    for {
+      event <- getEvent(eventId)
+      fsbTypes <- getFsbTypes
+      maybeFsbType <- Future(fsbTypes.find(f => f.key == event.description))
+      fsbType: FsbType = maybeFsbType.getOrElse(throw new Exception(s"FsbType with event description ${event.description} not found"))
```

Comment:
Can you combine these two lines (maybeFsbType and fsbType)? I think if it goes wrong you'll end in a failed future, with the exception inside